src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs

```diff
@@ -6,31 +6,47 @@
 using System.Linq;
 using Microsoft.Azure.WebJobs.Script.Config;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
 {
-    public class DiagnosticEventLogger : ILogger
+    public class DiagnosticEventLogger : ILogger, IDisposable
     {
         private readonly IDiagnosticEventRepositoryFactory _diagnosticEventRepositoryFactory;
         private readonly IEnvironment _environment;
+        private readonly IDisposable _standbyChangeHandler;
+
         private IDiagnosticEventRepository _diagnosticEventRepository;
+        private bool _isEnabled;
 
-        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment)
+        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment,
+            IOptionsMonitor<StandbyOptions> standbyOptions)
         {
             _diagnosticEventRepositoryFactory = diagnosticEventRepositoryFactory;
             _environment = environment;
+
+            SetEnabledState(standbyOptions.CurrentValue);
+
+            // Re-evaluate whether this is enabled after specialization
+            if (standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyChangeHandler = standbyOptions.OnChange(o => SetEnabledState(o));
+            }
         }
 
-        public IDisposable BeginScope<TState>(TState state)
+        private void SetEnabledState(StandbyOptions options)
         {
-            return null;
+            _isEnabled = !options.InStandbyMode &&
+                !FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDiagnosticEventLogging, _environment);
         }
 
-        public bool IsEnabled(LogLevel logLevel)
+        public IDisposable BeginScope<TState>(TState state)
```

Comment:
Do we need this method if all it does is return null? Or is there more to it?

Reply:
I think we need it since we are implementing `ILogger`